lib/utils/getFormatter.mjs

```diff
@@ -0,0 +1,58 @@
+import { EXIT_CODE_INVALID_CONFIG } from '../constants.mjs';
+import formatters from '../formatters/index.mjs';
+import getConfigForFile from '../getConfigForFile.mjs';
+import getFormatterOptionsText from './getFormatterOptionsText.mjs';
+
+/** @import {Formatter, FormatterType, InternalApi} from 'stylelint' */
+
+/**
+ * @param {InternalApi} stylelint
+ * @param {Object} [params]
+ * @param {FormatterType | Formatter | undefined} [params.formatterType]
+ * @param {string} [params.filePath]
+ * @returns {Promise<Formatter>}
+ */
+export default async function getFormatter(stylelint, params = {}) {
+	const { formatterType, filePath } = params;
+	const cwd = stylelint._options.cwd;
+	const configPath = stylelint._options.configFile || filePath || cwd;
+	let formatter = formatterType;
+
+	if (!formatter) {
+		let configForFile;
+
+		try {
+			configForFile = await getConfigForFile(stylelint, configPath, filePath);
+		} catch (err) {
+			if (err instanceof Error && 'code' in err && err.code === EXIT_CODE_INVALID_CONFIG) {
+				configForFile = undefined;
+			} else {
+				throw err;
```

Comment:
This should have a message.
e.g. `throw new Error('foo bar qux');`

Reply:
I don't think it's necessary to throw the original `err`.